### PR TITLE
[pvr] Combines PR #6333 and #6685

### PR DESCRIPTION
**Pull request by xhaggi**
This PR is used to merge #6333, #6685 and the necessary API bump.

### CODE REVIEW

**Comment by xhaggi** [comment]
jenkins build this please

**Comment by xhaggi** [comment]
jenkins build this please

**Comment by xhaggi** [comment]
build failure is unrelated.

**Comment by FernetMenta** [comment]
why is an api change without announcement in the forum mergered that fast? all addons are broken now, right?

**Comment by xhaggi** [comment]
Do we have a new guideline for merging PR's that introduce PVR API changes?

**Comment by opdenkamp** [comment]
well, didn't we decide to do this during alphas and have authors fix them up? that's the main difference between "old style" and now, so we don't have to do a ton of PRs just to do one api change?
it should of course be announced on the forums now that we don't do it ourselves anymore. there's a sticky forum thread for it

**Comment by FernetMenta** [comment]
yes,we did. i am just saing that giving addon devs the info a couple of days in advance would be nice

File: xbmc/epg/EpgDatabase.cpp [hunk 1]
```diff
@@ -99,6 +105,17 @@ void CEpgDatabase::UpdateTables(int iVersion)
 
   if (iVersion < 9)
     m_pDS->exec("ALTER TABLE epgtags ADD sIconPath varchar(255);");
+  
+  if (iVersion < 10)
```
Comment (Montellese) [inline xbmc/epg/EpgDatabase.cpp]: This breaks updating of the Epg database from version 9 to version 10 with SQLite. I get the following error message:
```
SQL: SQL error or missing database
Query: ALTER TABLE epgtags ADD sOriginalTitle  varchar(128), sCast           varchar(255), sDirector       varchar(255), sWriter         varchar(255), iYear           integer, sIMDBNumber     varchar(50);
```

File: xbmc/GUIInfoManager.cpp [hunk 2]
```diff
@@ -4783,20 +4840,53 @@ std::string CGUIInfoManager::GetItemLabel(const CFileItem *item, int info, std::
...
+        iSeason = item->GetEPGInfoTag()->SeriesNumber();
+        iEpisode = item->GetEPGInfoTag()->EpisodeNumber();
+      }
+      else if (item->HasVideoInfoTag())
+      {
+        iSeason = item->GetVideoInfoTag()->m_iSeason;
+        iEpisode = item->GetVideoInfoTag()->m_iEpisode;
+      }
+
+      if (iEpisode > 0)
```
Comment (Montellese) [inline xbmc/GUIInfoManager.cpp]: `iEpisode` may be used uninitialized here (same goes for `iSeason` two lines further down).

File: xbmc/GUIInfoManager.cpp [hunk 3]
```diff
@@ -4783,20 +4840,53 @@ std::string CGUIInfoManager::GetItemLabel(const CFileItem *item, int info, std::
...
+        CEpgInfoTagPtr tag(item->GetPVRChannelInfoTag()->GetEPGNow());
+        if (tag)
+          iSeason = tag->SeriesNumber();
+      }
+      else if (item->HasEPGInfoTag())
+        iSeason = item->GetEPGInfoTag()->SeriesNumber();
+      else if (item->HasVideoInfoTag())
+        iSeason = item->GetVideoInfoTag()->m_iSeason;
+
+      if (iSeason > 0)
```
Comment (Montellese) [inline xbmc/GUIInfoManager.cpp]: `iSeason` may be used uninitialized here

**Comment by Montellese** [comment]
Did anyone review this?

**Comment by xhaggi** [comment]
sure if you take a look at the initial PR. will get my hands on asap.

File: xbmc/epg/EpgDatabase.cpp (hunk 1 above)
Comment (xhaggi) [inline xbmc/epg/EpgDatabase.cpp]: WTF grrrr I hate sqlite and its limited SQL and nobody cares about at review. Will push a fix for both asap. Thanks for pointing this out.

Comment (Montellese) [inline xbmc/epg/EpgDatabase.cpp]: Hehe well IMO SQLite only has limited SQL support when it comes to table management but it supports a lot more stuff when it comes to sub-selects, exists etc.
As most people will be using an SQLite database I always develop using SQLite. It's also much easier to backup/drop during development than MySQL.

**Comment by xhaggi** [comment]
@FernetMenta I've updated the "PVR API changes" thread http://forum.kodi.tv/showthread.php?tid=97764&pid=1957562#pid1957562

**Pull request merged**